packages/test/test-end-to-end-tests/src/test/serializeAfterFailedAttach.spec.ts

```diff
@@ -0,0 +1,219 @@
+
+import { strict as assert } from "assert";
+import * as semver from "semver";
+import { AttachState, IContainer, IFluidCodeDetails } from "@fluidframework/container-definitions";
+import { Loader } from "@fluidframework/container-loader";
+import {
+	LocalCodeLoader,
+	TestFluidObjectFactory,
+	ITestFluidObject,
+	TestFluidObject,
+	createDocumentId,
+	LoaderContainerTracker,
+	ITestObjectProvider,
+} from "@fluidframework/test-utils";
+import { IContainerRuntimeBase } from "@fluidframework/runtime-definitions";
+import { IFluidHandle, IRequest } from "@fluidframework/core-interfaces";
+import { describeCompat } from "@fluid-private/test-version-utils";
+import { IDocumentServiceFactory } from "@fluidframework/driver-definitions";
+import type { ISharedMap } from "@fluidframework/map";
+import { wrapObjectAndOverride } from "../mocking.js";
+
+describeCompat(
+	`Serialize After Failure to Attach Container Test`,
+	"NoCompat",
+	(getTestObjectProvider, apis) => {
+		const { SharedMap, SharedString } = apis.dds;
+
+		const codeDetails: IFluidCodeDetails = {
+			package: "detachedContainerTestPackage1",
+			config: {},
+		};
+		const sharedStringId = "ss1Key";
+		const sharedMapId = "sm1Key";
+
+		let provider: ITestObjectProvider;
+		let loader: Loader;
+		let request: IRequest;
+		const loaderContainerTracker = new LoaderContainerTracker();
+
+		async function createDetachedContainerAndGetEntryPoint() {
+			const container: IContainer = await loader.createDetachedContainer(codeDetails);
+			// Get the root dataStore from the detached container.
+			const defaultDataStore = (await container.getEntryPoint()) as TestFluidObject;
+			return {
+				container,
+				defaultDataStore,
+			};
+		}
+
+		function createTestLoader(): Loader {
+			const factory: TestFluidObjectFactory = new TestFluidObjectFactory([
+				[sharedStringId, SharedString.getFactory()],
+				[sharedMapId, SharedMap.getFactory()],
+			]);
+			const codeLoader = new LocalCodeLoader([[codeDetails, factory]], {});
+			const testLoader = new Loader({
+				urlResolver: provider.urlResolver,
+				documentServiceFactory: wrapObjectAndOverride<IDocumentServiceFactory>(
+					provider.documentServiceFactory,
+					{
+						createContainer: () => assert.fail("fail on attach"),
+					},
+				),
+				codeLoader,
+				logger: provider.logger,
+				configProvider: {
+					getRawConfig: (name) =>
+						name === "Fluid.Container.RetryOnAttachFailure" ? true : undefined,
+				},
+			});
+			loaderContainerTracker.add(testLoader);
+			return testLoader;
+		}
+
+		const createPeerDataStore = async (containerRuntime: IContainerRuntimeBase) => {
+			const dataStore = await containerRuntime.createDataStore(["default"]);
+			const peerDataStore = (await dataStore.entryPoint.get()) as ITestFluidObject;
+			return {
+				peerDataStore,
+				peerDataStoreRuntimeChannel: peerDataStore.channel,
+			};
+		};
+
+		beforeEach("createLoader", async function () {
+			provider = getTestObjectProvider();
+			if (
```

Comment:
i don't think this skip is necessary

Reply:
i would actually just get rid of this beforeEach completely, and have each test manages its own loader/request. i think it makes the tests easier to read and debug, especially you aren't the author, or haven't used them in a long time

Reply:
createDetachedContainerAndGetEntryPoint will also need to  be modified to take an IHostLoader, so tests can pass theirs in